### PR TITLE
Rename AutoCloseableExecutorService, shutdown executor immediately, e…

### PR DESCRIPTION
**Pull request by johnou**
…xtract thread factory into util class.

### CODE REVIEW

File: src/main/java/io/github/lukehutch/fastclasspathscanner/FastClasspathScanner.java [hunk 1]
```diff
@@ -178,7 +177,6 @@ private synchronized ScanSpec getScanSpec() {
                 try {
                     // for (int i = 0; i < 10000000; i++) { // TODO:
```
Comment (johnou) [inline src/main/java/io/github/lukehutch/fastclasspathscanner/FastClasspathScanner.java]: An interesting todo :)

File: src/main/java/io/github/lukehutch/fastclasspathscanner/FastClasspathScanner.java [hunk 2]
```diff
@@ -219,32 +217,22 @@ public synchronized static final String getVersion() {
...
-                    });
+                    threadFactory);
         }
 
         /** Shut down thread pool on close(). */
         @Override
         public void close() {
             try {
-                shutdown();
+                shutdownNow();
```
Comment (johnou) [inline src/main/java/io/github/lukehutch/fastclasspathscanner/FastClasspathScanner.java]: from what I could see all operations inside the try catch for AutoCloseableExecutorService are blocking so this should be safe?

**Comment by johnou** [comment]
@lukehutch finally managed to look at the master branch using a monitor, awesome work!

File: src/main/java/io/github/lukehutch/fastclasspathscanner/FastClasspathScanner.java (hunk 1 above)
Comment (lukehutch) [inline src/main/java/io/github/lukehutch/fastclasspathscanner/FastClasspathScanner.java]: Oh yeah, this was for profiling / performance testing :)

File: src/main/java/io/github/lukehutch/fastclasspathscanner/FastClasspathScanner.java (hunk 2 above)
Comment (lukehutch) [inline src/main/java/io/github/lukehutch/fastclasspathscanner/FastClasspathScanner.java]: That was the intent, yes.

**Comment by lukehutch** [comment]
Thanks for your work on this!

**Pull request merged**